src/mono/mono/mini/mini.h

```diff
@@ -2998,7 +2999,7 @@ mini_class_is_simd (MonoCompile *cfg, MonoClass *klass)
 		return TRUE;
 	int size = mono_type_size (m_class_get_byval_arg (klass), NULL);
 #ifdef TARGET_ARM64
-	if (size == 8 || size == 16)
+	if (size == 8 || size == 16 || (size == 12 && !strcmp (m_class_get_name (klass), "Vector3")))
```

Comment:
Why do we need the additional condition `&& !strcmp (m_class_get_name (klass), "Vector3"))` here?
Isn't `size == 12` sufficient?